src/main/java/uk/gov/hmcts/dts/fact/entity/CourtAddress.java

```diff
@@ -28,4 +29,29 @@
     private String townName;
     private String townNameCy;
     private String postcode;
+
+    public CourtAddress(final Court court, final AddressType addressType, final List<String> addressLines, final List<String> addressLinesCy,
+                        final String townName, final String townNameCy, final String postcode) {
+        this.court = court;
+        this.addressType = addressType;
+        this.address = convertAddressLines(addressLines);
+        this.addressCy = convertAddressLines(addressLinesCy);
+        this.townName = townName;
+        this.townNameCy = townNameCy;
+        this.postcode = postcode;
+    }
+
+    private String convertAddressLines(final List<String> addressLines) {
+        if (!CollectionUtils.isEmpty(addressLines)) {
+            final StringBuilder builder = new StringBuilder();
+            final int lastLine = addressLines.size() - 1;
+            for (int i = 0; i < lastLine; i++) {
+                builder.append(addressLines.get(i))
+                    .append(System.lineSeparator());
+            }
+            builder.append(addressLines.get(lastLine));
+            return builder.toString();
+        }
+        return null;
```

Comment:
Do we need to return null here? Can we return an Optional<String> if we need to do so? And if it is empty, is this normally the case? If it isn't would a logging message helpful as a warning, or a debug? Btw your hair looks lovely today ofc

Reply:
I think optional is not right in this case. This is a private method just used by the constructor again. I modified it slightly to return a null string for empty list